internal/services/sql/sql_managed_instance_failover_group_resource_test.go

```diff
@@ -0,0 +1,370 @@
+package sql_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/sql/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type SqlInstanceFailoverGroupResource struct{}
+
+func TestAccAzureRMSqlInstanceFailoverGroup_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sql_managed_instance_failover_group", "test")
+	r := SqlInstanceFailoverGroupResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: SqlManagedInstanceResource{}.dnsZonePartner(data),
+		},
+		{
+			// It speeds up deletion to remove the explicit dependency between the instances
+			Config: SqlManagedInstanceResource{}.emptyDnsZonePartner(data),
+		},
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			// disconnect
+			Config: SqlManagedInstanceResource{}.emptyDnsZonePartner(data),
+		},
+	})
+}
+
+func TestAccAzureRMSqlInstanceFailoverGroup_change(t *testing.T) {
```

Comment:
This tests isn't necessary anymore since we're recreating the resource when we change the partner managed instance id

Reply:
Removed 🥳